ops/testing.py

```diff
@@ -1546,7 +1546,11 @@ def make_dir(
             )
 
     def remove_path(self, path: str, *, recursive: bool = False):
-        file_or_dir = self._fs.get_path(path)
+        try:
+            file_or_dir = self._fs.get_path(path)
+        except FileNotFoundError:
+            return
```

Comment:
See updated comment [here](https://github.com/canonical/operator/issues/676#issuecomment-1009399323) -- I believe this should still raise if recursive is False, so we need something like:

```
try:
    file_or_dir = self._fs.get_path(path)
except FileNotFoundError:
    if recursive:
        return  # Pebble doesn't give not-found error when recursive is specified
    else:
        raise
```

Reply:
I think the test will need to be update accordingly. This also emphasizes the need to have a mode to run the filesystem tests against real pebble. See `TestRealPebble` in `test/test_pebble.py`.

Reply:
Thanks for the help! Updated.